tensorflow.go

```diff
@@ -0,0 +1,172 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+	tf "github.com/tensorflow/tensorflow/tensorflow/go"
+)
+
+type Session struct {
+	session tf.TF_Session
+}
+
+func NewSession() (*Session, error) {
+	status := tf.TF_NewStatus()
+	return &Session{
+		session: tf.TF_NewSession(
+			tf.TF_NewSessionOptions(),
+			status,
+		),
+	}, statusToError(status)
+}
+
+type Tensor struct {
+	tensor tf.TF_Tensor
+	buf    []byte
+}
+
+type TensorShape [][]int64
+
+var (
+	TensorShapeScalar = TensorShape{{1}}
+)
+
+func NewTensor(dataType tf.DataType, shape TensorShape, data interface{}) (*Tensor, error) {
+	// TODO(tmc): ensure data is a slice
+	v := reflect.ValueOf(data)
+	if v.Kind() != reflect.Slice {
+		return nil, fmt.Errorf("tensorflow: 'data' argument must be a slice")
+	}
+	dataSize := v.Len() * int(v.Type().Elem().Size())
+	dataPtr := v.Pointer()
+	return newTensor(dataType, shape, dataPtr, dataSize)
+}
+
+func newTensor(dataType tf.DataType, shape TensorShape, data uintptr, size int) (*Tensor, error) {
+	t := &Tensor{
+		tensor: tf.TF_NewTensor_wrapper(tf.TF_DataType(dataType), &(shape[0][0]), len(shape), data, int64(size)),
+	}
+
+	return t, nil
+}
+
+func encodeStrings(in []string) []byte {
```

Comment:
This function could use a unit test.  (In general, I'd love to see more unit tests throughout the Go bindings.)